.github/workflows/e2e-intake.yml

```diff
@@ -0,0 +1,332 @@
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - name: Cache Playwright browsers
+        id: playwright-cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.PLAYWRIGHT_CACHE_PATH }}
+          key: ${{ runner.os }}-playwright-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-playwright-
+
+      - name: Try restore user.json cache
+        id: cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.USER_JSON_PATH }}
+          key: ${{ env.USER_JSON_CACHE_KEY }}
+
+      - name: Check out secrets repo
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.SECRETS_REPO }}
+          ssh-key: ${{ secrets.DEPLOY_OTTEHR_KEY }}
+          path: 'secrets'
+
+      - name: Setup environment
+        run: |
+          setup_env() {
+            mkdir -p "packages/zambdas/.env"
+            cp secrets/zambdas/* "packages/zambdas/.env"
+            mkdir -p "apps/$app/env"
+            cp "secrets/$app/app/tests."*.json "apps/$app/env/"
+            cp "secrets/$app/app/.env"* "apps/$app/env/"
+          }
+      
+          for app in ehr intake; do
+            setup_env "$app"
+          done
+
+      - name: Install dependencies
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - name: Install Playwright browsers
+        run: npx playwright install chromium
+
+      - name: Run login
+        id: login
+        run: npm run intake:e2e:${{ github.event.inputs.environment || 'local' }}:login
+        env:
+          CI: true
+
+      - name: Validate user.json
+        if: steps.login.outcome == 'success'
+        id: validate
+        run: |
+          node -e '
+          const fs = require("fs");
+          const path = "${{ env.USER_JSON_PATH }}";
+
+          if (!fs.existsSync(path)) {
+            console.log(`File ${path} does not exist`);
+            process.exit(1);
+          }
+          
+          const data = JSON.parse(fs.readFileSync(path, "utf8"));
+          
+          const hasAuth0Token = data.origins?.some(origin => 
+            origin.localStorage && 
+            origin.localStorage.some(item => item.name?.includes?.("auth0"))
+          );
+          
+          if (!hasAuth0Token) {
+            console.log("No Auth0 token found");
+            process.exit(1);
+          }
+          
+          console.log("User.json validated successfully");
+          '
+
+      - name: Save user.json to cache
+        if: steps.validate.outcome == 'success'
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.USER_JSON_PATH }}
+          key: ${{ env.USER_JSON_CACHE_KEY }}
+
+  intake-e2e-tests:
+    needs: [check-token-validity, intake-login]
+    if: |
+      always() && !cancelled() && 
```

Comment:
do we need the `always()` if there are other conditions?

Reply:
always() is needed to run tests if the login step is skipped